### PR TITLE
The error status is inside err.cause

### PR DESCRIPTION
**Pull request by k-thornton**
This one's not quite as clean, since it's not guaranteed that a failing loader actually has a status of 404.  It appears to be true for the existing loaders though, so it correctly catches the error.

A cleaner way to do this might be to have the loaders themselves decide when to fail quietly and when to throw an error.  If it's simply not finding the ABI, fail quietly instead.

https://github.com/shazow/whatsabi/issues/159

### CODE REVIEW

File: src/loaders.ts [hunk 1]
```diff
@@ -121,7 +121,7 @@ export class MultiABILoader implements ABILoader {
                     return r;
                 }
             } catch (err: any) {
-                if (err.status === 404) continue;
+                if (err.cause?.status === 404) continue;
```
Comment (shazow) [inline src/loaders.ts:124]: HMM I could have sworn there was some loader that threw an error that had a status property, but maybe I'm wrong?

Maybe we need a test that goes over empty result conditions for each loader we have.

What do you think? Are you into it or should I do it? :)

**Comment by shazow** [comment]
> A cleaner way to do this might be to have the loaders themselves decide when to fail quietly and when to throw an error. 

I'm into that if you're interested in making the change, or could do it in a separate PR later.

We've generally been moving towards doing better at wrapping errors closer to the source, so that would be aligned with that effort.

File: src/loaders.ts (hunk 1 above)
Comment (shazow) [reply src/loaders.ts:124]: @k-thornton Do you remember under what condition you were able to trigger a 404 status on our loaders?

I'm trying to add a test for it but haven't been able to reproduce yet. https://github.com/shazow/whatsabi/pull/161/files#diff-2d7e35d8a3a306281e2352b54fe4ec41b741e4a6bfe9d1233670ddf569b6b4e4R230-R244

Comment (shazow) [reply src/loaders.ts:124]: Ah looks like it's just the new Blockscout loader when using loadABI, fixing it.

**Comment by shazow** [comment]
Fixed via #163
